libdrgn/drgn.h.in

```diff
@@ -817,6 +817,24 @@ struct drgn_error *drgn_program_find_symbol_by_name(struct drgn_program *prog,
 						    const char *name,
 						    struct drgn_symbol **ret);
 
+/**
+ * Get all global and local symbols, optionally corresponding to the given name.
+ *
+ * @param[in] prog Program.
+ * @param[in] name Name to match. If @c NULL, returns all symbols.
+ * @param[out] syms_ret Pointer to a variable to be filled with an array of symbols.
+ * The resulting value is non-@c NULL on success, and has a length of
+ * @c *count_ret pointers. Each symbol shuld be freed with @ref
+ * drgn_symbol_destroy(), and the array itself should be freed with the @ref
+ * drgn_symbols_free().
+ * @param[out] count_ret Returned number of symbols in @p syms_ret.
```

Comment:
We probably shouldn't mention freeing the individual symbols, because that is handled internally by `drgn_symbols_free()`. Also, let's call it `drgn_symbols_destroy()` for consistency with the rest of the libdrgn API.

```suggestion
 * @param[out] syms_ret Returned array of symbols. On success, this must be
 * freed with @ref drgn_symbols_destroy().
 * @param[out] count_ret Returned number of symbols in @p syms_ret.
```